demo/bin/example.gdextension

```diff
@@ -5,12 +5,12 @@ compatibility_minimum = "4.1"
 
 [libraries]
 
-macos.debug = "res://bin/macos/macos.framework/libEXTENSION-NAME.macos.template_debug"
-macos.release = "res://bin/macos/macos.framework/libEXTENSION-NAME.macos.template_release"
-ios.debug = "res://bin/ios/ios.framework/libEXTENSION-NAME.ios.template_debug"
-ios.release = "res://bin/ios/ios.framework/libEXTENSION-NAME.ios.template_release"
-windows.debug.x86_32 = "res://bin/windows/libEXTENSION-NAME.windows.template_debug.x86_32.dll"
-windows.release.x86_32 = "res://bin/windows/libEXTENSION-NAME.windows.template_release.x86_32.dll"
+macos.debug = "res://bin/macos/EXTENSION-NAME-macos.framework/libEXTENSION-NAME.macos.template_debug.universal.dylib"
```

Comment:
- Files in the framework should not have `.dylib` extensions. Standard framework structure always have binary with exactly the same as the framework name, e.g, `EXTENSION-NAME.framework\EXTENSION-NAME`.
- There should be two separate frameworks for `template_debug` and `template_release`.
- Frameworks should always have `Resources/Info.plist` in them (it is present in the main repo, not sure why it's missing here - [Info.plist](https://github.com/godotengine/godot-cpp/blob/master/test/project/bin/libgdexample.macos.template_debug.framework/Resources/Info.plist)).
- `.gdextension` should point to the `.framework` folder, not to the file in it.

Reply:
I had to look it up, you're correct: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html#//apple_ref/doc/uid/20002253-BAJEJJAB

I'll adjust the PR to use the macOS recommended .framework structure.

Reply:
Actually, before i proceed much further: Has there been discussion around whether to use a .framework or .dylib structure for gdextensions?

Both can be signed, and the main difference is that .framework structures can contain multiple versions and sub dylibs. I'm not sure we even want this for gdextensions?

Reply:
Both can be used for macOS, for iOS it's only `.framework`, raw `.dylib`s are not accepted by App Store.